benchmarks/ASQA/asqa_benchmark.py

```diff
@@ -22,19 +22,20 @@ def __init__(self) -> None:
         """Initialization."""
         super().__init__()
 
-    def prepare_data(self, label_column: str, input_column: str):
-        """Modify self.dataset for different metric.
+    def prepare_data(self, metric: str, label_column: str, input_column: str):
+        """Modify self.dataset for different metric. Remove the existing metric column for metric to be evaluated and add the label column will be uesd.
 
         Args:
             input_column: The column name of the input text that has already existed in self.dataset, e.g. `long_answer`.
             label_column: The column name of the label text that the metric requires, e.g. `gt_answer`.
         """
+        if metric in self.dataset.column_names:
+            self.dataset = self.dataset.remove_columns(metric)
+
         if input_column not in self.dataset.column_names:
             raise ValueError(f"The input column {input_column} is not in the dataset. Please check the column names.")
 
-        if label_column in self.dataset.column_names:
-            self.dataset = self.dataset.remove_columns(label_column)
-        self.dataset = self.dataset.add_column(label_column, self.dataset[input_column])
+        self.dataset = self.dataset.map(lambda example: {label_column: example[input_column]}, batched=True)
```

Comment:
I think you are trying to rename the dataset column from input name to label name. why not do:
self.dataset.rename_column("old_column_name", "new_column_name")?

Reply:
Fixed.